### PR TITLE
Fixes #23098 - Allow searching on all audited models by type

### PR DESCRIPTION
**Pull request by tbrisker**
This automatically adds all the audited models in an after_initilize
block so that models defined in plugins are also included. All the
search related definitions for audit have been extracted to a seperate
concern. Models using STI require special handling until audits are
fixed to properly work with STI.

### CODE REVIEW

File: app/models/concerns/audit_search.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      )
+    end
+
+    private
+
+    #This is a workaround needed until we get proper STI auditing
+    def audited_classes_without_sti
+      audited_class_names.reject do |name|
+        klass = name.constantize
+        name.include?(':') ||
```
Comment (ares) [inline app/models/concerns/audit_search.rb]: wouldn't this reject all namespaced classes, e.g. `ForemanOpenscap::ArfReport` or `Katello::HostCollection`? Also if this should remain, `::` would be more descriptive I think.

**Review by ares — CHANGES_REQUESTED** [review]
one question, otherwise looks good, I'll continue with testing meanwhile

File: app/models/concerns/audit_search.rb (hunk 1 above)
Comment (tbrisker) [reply app/models/concerns/audit_search.rb]: Good point, i'll fix this.

**Review by ares — APPROVED** [review]
Thanks, works nicely! I'll wait for Mr. Jenkins but good to be merged from my point of view.

**Comment by ares** [comment]
Thanks @tbrisker, merging!

**Pull request merged**